source/crud/tests/unified/estimatedDocumentCount.yml

```diff
@@ -265,3 +265,65 @@ tests:
                 count: *collection0Name
               commandName: count
               databaseName: *database0Name
+
+  - description: "estimatedDocumentCount with comment on 4.9.0 or greater"
+    runOnRequirements:
+      - minServerVersion: "4.9.0"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection0
+        arguments:
+          comment: &objectComment { key: "value"}
+        expectResult: 3
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                aggregate: *collection0Name
+                pipeline: *pipeline
```

Comment:
I assume this is necessary because of the `$collStats` approach discussed in [the CRUD spec](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#estimateddocumentcount).

I do think these comment tests would be better placed in their own file (e.g. `estimatedDocumentCount-comment.yml`), even if that means redefining the `$collStats` pipeline. That would be more consistent with existing tests, which tend to split out files for testing certain options.

Reply:
Good idea, thank. Extracted new tests to a separated file.